handler/todo_handler.go

```diff
@@ -0,0 +1,122 @@
+package handler
+
+import (
+	"fmt"
+	"go-todo-api/entities"
+	"go-todo-api/models"
+	"go-todo-api/repository"
+	"net/http"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+)
+
+type ToDoHandler struct {
+	ToDoRepository repository.ToDoRepositoryInterface
+}
+
+func NewToDoHandler(ToDoRepo repository.ToDoRepositoryInterface) *ToDoHandler {
+	return &ToDoHandler{ToDoRepository: ToDoRepo}
+}
+
+func (handler *ToDoHandler) PostToDo(c *gin.Context) {
+	var CreateToDo = &models.ToDoRequest{}
+
+	if err := c.BindJSON(&CreateToDo); err != nil {
+		fmt.Printf("Error: %v", err)
+		fmt.Println("")
```

Comment:
aşağıda boş print var ayrıca error yazdırmak için print değil logger kullanalım https://github.com/uber-go/zap